server/aws-lsp-codewhisperer/src/language-server/proxy-server.ts

```diff
@@ -7,76 +7,48 @@ import { QNetTransformServerToken } from './netTransformServer'
 import { QChatServer } from './qChatServer'
 import { QConfigurationServerToken } from './configuration/qConfigurationServer'
 import { readFileSync } from 'fs'
+import { ConfigurationOptions } from 'aws-sdk'
+import { HttpsProxyAgent } from 'hpagent'
 
-export const CodeWhispererServerTokenProxy = CodewhispererServerFactory(credentialsProvider => {
-    let additionalAwsConfig = {}
+const makeProxyConfig = () => {
+    let additionalAwsConfig: ConfigurationOptions = {}
     const proxyUrl = process.env.HTTPS_PROXY ?? process.env.https_proxy
-    const certs = process.env.AWS_CA_BUNDLE ? [readFileSync(process.env.AWS_CA_BUNDLE)] : undefined
 
     if (proxyUrl) {
-        const { getProxyHttpAgent } = require('proxy-http-agent')
-        const proxyAgent = getProxyHttpAgent({
+        const certs = process.env.AWS_CA_BUNDLE ? [readFileSync(process.env.AWS_CA_BUNDLE)] : undefined
+        const agent = new HttpsProxyAgent({
             proxy: proxyUrl,
             ca: certs,
         })
+
         additionalAwsConfig = {
-            httpOptions: proxyAgent,
+            httpOptions: {
```

Comment:
do we also need to set httpsOptions, as I see QChatServer setting both agents for http and https